examples~/quickstart/client/module_bindings/_Globals/SpacetimeDBClient.cs

```diff
@@ -10,59 +10,94 @@
 
 namespace SpacetimeDB.Types
 {
-	public enum ReducerType
+	public interface IReducerArgs : IReducerArgsBase
 	{
-		None,
-		SendMessage,
-		SetName,
+		bool InvokeHandler(EventContext ctx);
 	}
 
-	public interface IReducerArgs : IReducerArgsBase
+	public sealed class RemoteTables
 	{
-		ReducerType ReducerType { get; }
-		bool InvokeHandler(ReducerEvent reducerEvent);
+		public readonly RemoteTableHandle<EventContext, Message> Message = new();
+		public readonly RemoteTableHandle<EventContext, User> User = new();
 	}
 
-	public partial class ReducerEvent : ReducerEventBase
+	public sealed class RemoteReducers : RemoteBase<DbConnection>
 	{
-		public IReducerArgs? Args { get; }
+		internal RemoteReducers(DbConnection conn) : base(conn) {}
 
-		public string ReducerName => Args?.ReducerName ?? "<none>";
+		public delegate void SendMessageHandler(EventContext ctx, string text);
+		public event SendMessageHandler? OnSendMessage;
 
-		[Obsolete("ReducerType is deprecated, please match directly on type of .Args instead.")]
-		public ReducerType Reducer => Args?.ReducerType ?? ReducerType.None;
+		public void SendMessage(string text)
+		{
+			conn.InternalCallReducer(new SendMessageArgsStruct { Text = text });
+		}
 
-		public ReducerEvent(IReducerArgs? args) : base() => Args = args;
-		public ReducerEvent(TransactionUpdate dbEvent, IReducerArgs? args) : base(dbEvent) => Args = args;
+		public bool InvokeSendMessage(EventContext ctx, SendMessageArgsStruct args)
+		{
+			if (OnSendMessage == null) return false;
+			OnSendMessage(
+				ctx,
+				args.Text
+			);
+			return true;
+		}
+		public delegate void SetNameHandler(EventContext ctx, string name);
+		public event SetNameHandler? OnSetName;
 
-		[Obsolete("Accessors that implicitly cast `Args` are deprecated, please match `Args` against the desired type explicitly instead.")]
-		public SendMessageArgsStruct SendMessageArgs => (SendMessageArgsStruct)Args!;
-		[Obsolete("Accessors that implicitly cast `Args` are deprecated, please match `Args` against the desired type explicitly instead.")]
-		public SetNameArgsStruct SetNameArgs => (SetNameArgsStruct)Args!;
+		public void SetName(string name)
+		{
+			conn.InternalCallReducer(new SetNameArgsStruct { Name = name });
+		}
+
+		public bool InvokeSetName(EventContext ctx, SetNameArgsStruct args)
+		{
+			if (OnSetName == null) return false;
+			OnSetName(
+				ctx,
+				args.Name
+			);
+			return true;
+		}
+	}
+
+	public partial class EventContext : EventContextBase<RemoteTables, RemoteReducers>
+	{
+		public IReducerArgs? Args { get; }
+
+		public string ReducerName => Args?.ReducerName ?? "<none>";
```

Comment:
This is not as designed - `EventContext` should have a single field `event: Event<Reducer>` (where `Reducer` is an enum / tagged union defined in the codegen).

Reply:
Yes, those parts of the proposal are not covered in this PR - this is struct as it looked before. Jeremie is working on other parts separately.

Reply:
Can y'all update the PR title and description to reflect that, then? Puru was confused by this and wasted time in his implementation of the TypeScript version because he was basing it off this PR.

Reply:
Yeah I was a bit confused by it too, see my comment above. https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/131#discussion_r1771382419

@lcodes can you update the PR as suggested to clarify it's only implementing the DbConnection bits?